### ISSUE TITLE
Are there any tests that actually work?

### ISSUE THREAD

**Issue by migerh** [wmonk/create-react-app-typescript#47]
Like @jeremistadler I was trying to update this to TypeScript 2.2. Before submitting a PR I wanted to run some tests and found a few e2e shell scripts in /tasks. Are they supposed to work? I couldn't get them to run properly, too many eslint/tsc errors. I managed to fix a few of those but hit a wall so I thought I'd check back first if they are even supposed to work.

How do you test this project before updating the npm packages? I don't see how this would work without publishing the package. I supposed I could publish a scoped package but there's got to be a better way to test this…

**Comment by kevinjhanna** [comment]
To test a npm package manually you can use [npm link](https://docs.npmjs.com/cli/link).

```
cd your-package
npm link

# And then you cd to the directory you want to install the package
npm link your-package-name
```

The problem with `create-react-app-typescript` is that there is no package name declared in `package.json`, so this wouldn't work.

Is there a reason not to have a package name? The original `create-react-app` repo doesn't have one either.

**Comment by kevinjhanna** [comment]
To install a npm package locally without publishing you can use [npm link](https://docs.npmjs.com/cli/link).

```
cd your-package
npm link

# And then you cd to the directory you want to install the package
npm link your-package-name
```

The problem with `create-react-app-typescript` is that there is no package name declared in `package.json`, so this wouldn't work.

Is there a reason not to have a package name? The original `create-react-app` repo doesn't have one either.

**Comment by migerh** [comment]
I'm pretty sure I already tried that and it didn't work. But I just tried it again and saw that the script uses `npm pack` on the `react-scripts` package in the repo and then uses the generated tgz to bootstrap the test project. Thanks, that's exactly what I was looking for.

But is `npm run e2e` supposed to fail?

**Comment by wmonk** [comment]
Hey, currently none of the tests are meant to work. Some might, but I haven't invested time in making them work. I would happily accept a PR to get them working though, would definitely give a better indication before publishing. Currently i've been manually testing with a linked package.

**Comment by migerh** [comment]
I made some fixes in #49
